trunk/source/Library-ModelCheckerUtils/src/de/uni_freiburg/informatik/ultimate/lib/modelcheckerutils/cfg/IcfgProgramExecution.java

```diff
@@ -359,5 +361,12 @@ public IBacktranslationValueProvider<L, Term> getBacktranslationValueProvider()
 	public boolean isConcurrent() {
 		return mIsConcurrent;
 	}
+	
+	public void setOriginCfgScript(ManagedScript managedScript) {
+		mScript = managedScript;

```

Comment:
Is it necessary to make this class mutable?

Reply:
If `mScript` is used "on creation", as your comment above states, could you not set this in the constructor? (and make it a final field?)

Or, alternatively, pass the script alongside (but not as part of) the execution?